mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -105,6 +110,24 @@ static TerminatorTy verifyAndGetTerminator(Operation *op, Region &region,
   return nullptr;
 }
 
+/// Helper function to compute the difference between two values. This is used
+/// by the loop implementations to compute the trip count.
+static std::optional<llvm::APSInt> computeUbMinusLb(Value lb, Value ub,
```

Comment:
This is for a restricted form right? looks for ub = lb + const. E.g., this would fail in the case where ub and lb are constants.

Reply:
Yes it's not meant for constants.